compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2913,3 +2913,13 @@ class MatchTypeScrutineeCannotBeHigherKinded(tp: Type)(using Context)
   extends TypeMsg(MatchTypeScrutineeCannotBeHigherKindedID) :
     def msg(using Context) = i"the scrutinee of a match type cannot be higher-kinded"
     def explain(using Context) = ""
+
+class ToplevelDefCantBeInfix(sym: Symbol)(using Context)
+  extends SyntaxMsg(ToplevelDefCantBeInfixID):
+    def msg(using Context) = i"a toplevel $defName cannot be infix"
+    def explain(using Context) = ""
+    private val defName = 
+      if sym.flags.is(Method) then "def"
+      else if sym.flags.is(Mutable) then "var"
+      else if sym.flags.is(Given) then "given"
+      else "val"
```

Comment:
Missing end line break ^^'

VS Code tip:
There are settings called "Insert Final Newline" and "Trim Final NewLines" that automatically fix these for you